### PR TITLE
[BXMSPROD-2011] Added withKerberos utility method

### PR DESCRIPTION
**Pull request by lampajr**
**JIRA**: https://issues.redhat.com/browse/BXMSPROD-2011

Additional utility function that allows to execute closure within Kerberos authentication context

### CODE REVIEW

**Review by Ginxo — CHANGES_REQUESTED** [review]
thanks for this, please check my comments

File: vars/util.groovy [hunk 1]
```diff
@@ -601,4 +601,17 @@ String encode(String value, String encoding='UTF-8') {
...
+    withCredentials([file(credentialsId: keytabId, variable: 'KEYTAB_FILE')]) {
+        env.KERBEROS_PRINCIPAL = sh(returnStdout: true, script: ''' klist -kt $KEYTAB_FILE |grep REDHAT.COM | awk -F" " 'NR==1{print $4}' ''').trim()
+        def kerberosStatus = sh(returnStatus: true, script: "kinit ${env.KERBEROS_PRINCIPAL} -kt $KEYTAB_FILE")
+
+        if (kerberosStatus == 0) {
+            closure()
+        } else {
+            println "[ERROR] kinit failed with status $kerberosStatus"
+        }
+    }
```
[the remark this answers is not in the document]
Comment (lampajr) [reply vars/util.groovy:616]: Thanks @Ginxo , I totally agree with you!

[the remark this answers is not in the document]
Comment (lampajr) [reply vars/util.groovy]: Updated

**Review by Ginxo — APPROVED** [review]
LGTM

**Pull request merged**